### ISSUE TITLE
Cannot find methods for pausing/resuming speech

### ISSUE THREAD

**Issue by parhamdoustdar** [espeak-ng/espeak-ng#381]
Hi,

Screen readers provide this ability, so I'm sure I'm missing something, but there is no function defined in `speak_lib.h` that allows me to pause and resume speech.

The only way I can imagine this happening is asking espeak-ng to generate a wave file, and playing/pausing that instead, which, on first thought, doesn't seem like a sensible way to achieve this.

Is this the only way?

Thanks!

**Comment by rhdunn** [comment]
That functionality is not currently available in espeak itself. You could implement it by using the callback to track the progress through the text. Pausing would work like stopping. Playing could then adjust the text passed to espeak to start at the saved offset (e.g. the last clause start position).

I'll keep this open to look at adding support for this in espeak-ng directly.
